### PR TITLE
CS-1253 Refactor to use Secrets Manager

### PR DESCRIPTION
**Pull request by jwelborn-sugar**
for Sugar Credentials

### CODE REVIEW

File: src/core/api.js [hunk 1]
```diff
@@ -33,6 +32,9 @@ module.exports = () => {
         },
 
         call: async function(method, url, data, params) {
+            const secrets = JSON.parse(await Secrets);
+            const username = secrets.sugarUsername || '';
+            const password = secrets.sugarPass || '';
```
Comment (jwelborn-sugar) [inline src/core/api.js:37]: I moved this inside `call` so we can await the secrets. `module.exports` is synchronous by default, but `AWS.SecretsManager.getSecretValue` is async.

File: src/utils/aws/secrets.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            if ('SecretString' in data) {
+                secret = data.SecretString;
+            } else {
+                let buff = Buffer.from(data.SecretBinary, 'base64');
+                secret = buff.toString('ascii');
+            }
+            resolve(secret);
+        }
+    });
+});
```
Comment (jwelborn-sugar) [inline src/utils/aws/secrets.js:47]: I wrapped the code provided by AWS Secrets Manager in a promise so we can await the return of getSecretValue.

File: template.yml [hunk 3]
```diff
@@ -45,6 +45,16 @@ Resources:
...
+  # Secrets Manager Secret to store Sugar Username/Password
+  SugarSecrets:
+    Type: AWS::SecretsManager::Secret
+    Properties:
+      Name: SugarSecrets
+      Description: This secret stores your SugarCRM Credentials using AWS Secrets Manager
+      SecretString: !Sub 
+        - '{"sugarUsername": "${Username}", "sugarPass": "${Password}"}'
+        - { Username: !Ref SugarUsername, Password: !Ref SugarPassword }
+
```
Comment (jwelborn-sugar) [inline template.yml:57]: This syntax should be 1-1 compatible with CloudFormation templates. 🤞

File: template.yml [hunk 4]
```diff
@@ -55,8 +65,8 @@ Resources:
       Timeout: 100
       Environment:
         Variables:
-          sugarUsername: !Ref SugarUsername
-          sugarPass: !Ref SugarPassword
+          region: !Ref AWS::Region
+          secretManagerName: SugarSecrets
```
Comment (jwelborn-sugar) [inline template.yml:69]: The region and name of our SecretsManager secret store are needed in place of username/password so we can retrieve the credentials at runtime.

**Review by rogeryang-sugarcrm — APPROVED** [review]
Looks good to me

**Review by sugarhqi — APPROVED** [review]

**Review by sugarhqi — APPROVED** [review]

**Review by rogeryang-sugarcrm — APPROVED** [review]

**Pull request merged**